implementations/micrometer-registry-influx/src/main/java/io/micrometer/influx/InfluxMeterRegistry.java

```diff
@@ -283,4 +414,40 @@ public String toString() {
         }
     }
 
+    private enum InfluxDBVersion {
+        V1 {
+            @Override
+            String writeEndpoint(final InfluxMeterRegistry registry) {
+                InfluxConfig config = registry.config;
+                return config.uri() + "/write?consistency=" + config.consistency().toString().toLowerCase() + "&precision=ms&db=" + config.db();
+            }
+
+            @Override
+            void addHeaderToken(final InfluxMeterRegistry registry, final HttpSender.Request.Builder requestBuilder) {
+                InfluxConfig config = registry.config;
+                if (config.token() != null) {
+                    requestBuilder.withHeader("Authorization", "Bearer " + config.token());
+                }
+            }
+        },
+        V2 {
+            @Override
+            String writeEndpoint(final InfluxMeterRegistry registry) throws UnsupportedEncodingException {
+                InfluxConfig config = registry.config;
+                return config.uri() + "/api/v2/write?&precision=ms&bucket=" + URLEncoder.encode(config.bucket(), "UTF-8") + "&org=" + URLEncoder.encode(registry.org, "UTF-8");
```

Comment:
InfluxDB supports variable timestamp precision. Looking at the Micrometer documentation this doesn't seem to be an option?

Reply:
Yeah. Currently the Micrometer only supports millisecond precision for timestamp. See - https://github.com/micrometer-metrics/micrometer/blob/07701bd311af40419964a54add6ae60b3cfdf29b/micrometer-core/src/main/java/io/micrometer/core/instrument/Clock.java#L47